File: Sprinters.Webshop.BFF/Sprinters.Webshop.BFF/Entities/Bestelling.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Sprinters.SharedTypes.BeheerService.Entities;

namespace Sprinters.Webshop.BFF.Entities
{
    public class Bestelling
    {
        public int Id { get; set; }

        [StringLength(255, ErrorMessage = "Adresregel1 mag niet meer dan 255 karakters bevatten")]
        [Required(ErrorMessage = "AdresRegel1 is verplicht")]
        public string AdresRegel1 { get; set; }

        public string AdresRegel2 { get; set; }

        [StringLength(255, ErrorMessage = "Plaats mag niet meer dan 255 karakters bevatten")]
        [Required(ErrorMessage = "Plaats is verplicht")]
        public string Plaats { get; set; }

        [Required(ErrorMessage = "Postcode is verplicht")]
        [StringLength(7, ErrorMessage = "Postcode mag niet langer zijn dan 7 karakters")]
        public string Postcode { get; set; }

        public Klant Klant { get; set; }
        public string KlantId { get; set; }

        public BestellingStatus BestellingStatus { get; set; }

        public List<BestellingItem> BesteldeArtikelen { get; set; }
    }
}

File: Sprinters.Magazijn.Angular/src/app/models/bestelling.ts
import { BesteldeArtikelen } from './BesteldeArtikelen';
import { Klant } from './Klant';

export class Bestelling {
    id: number;
    voornaam: string;
    achternaam: string;
    adres: string;
    adresRegel2?: any;
    plaats: string;
    postcode: string;
    inBehandeling: boolean;
    besteldeArtikelen: BesteldeArtikelen[];
    klant?: Klant;

}
